File: packages/urdu-js/tests/analysis/script.test.ts
import { describe, it, expect } from 'vitest'
import { getScript, getUrduDensity } from '../../src/analysis/script.js'

describe('getScript', () => {
  it('empty → unknown', () => expect(getScript('')).toBe('unknown'))
  it('Urdu text → urdu', () => expect(getScript('پاکستان')).toBe('urdu'))
  it('Latin only → latin', () => expect(getScript('Hello World')).toBe('latin'))
  it('spaces only → unknown', () => expect(getScript('   ')).toBe('unknown'))
})

describe('getUrduDensity', () => {
  it('empty → 0', () => expect(getUrduDensity('')).toBe(0))
  it('pure Urdu > 0', () => expect(getUrduDensity('پاکستان')).toBeGreaterThan(0))
  it('pure Latin → 0', () => expect(getUrduDensity('Hello')).toBe(0))
})

File: packages/urdu-dart/test/analysis/script_test.dart
import 'package:test/test.dart';
import 'package:urdu_tools/urdu_tools.dart';

void main() {
  group('getScript', () {
    test('empty -> unknown',
        () => expect(getScript(''), equals(Script.unknown)));
    test('Urdu text -> urdu',
        () => expect(getScript('پاکستان'), equals(Script.urdu)));
    test('Latin only -> latin',
        () => expect(getScript('Hello World'), equals(Script.latin)));
    test('spaces only -> unknown',
        () => expect(getScript('   '), equals(Script.unknown)));
  });

  group('getUrduDensity', () {
    test('empty -> 0', () => expect(getUrduDensity(''), equals(0)));
    test('pure Urdu > 0',
        () => expect(getUrduDensity('پاکستان'), greaterThan(0)));
    test('pure Latin -> 0', () => expect(getUrduDensity('Hello'), equals(0)));
  });
}
